### DISCUSSION TITLE
Searching for help with RAC dialog

### DISCUSSION THREAD

**Discussion by p1pah** [adobe/react-spectrum#5952] [Q&A] [answered]
Good afternoon, 

We have recently converted to RAC and have been loving it. The only hiccup in our app is when trying to use AGGrid and the RAC dialog together. In the attached code sandbox you will see that AGGrid works perfectly fine out side of a "dialog component" But when using it in a dialog, the tabbing gets all out of whack. Is this potentially a tabIndex issue or something else? We have had to use another dialog solution because of this, but any help/insight would be greatly appreciated!

https://codesandbox.io/p/sandbox/rac-cqlk4n

**Comment by snowystinger** [comment]
Looks like they've implemented a roving tab index, there might be interference with our containing FocusScope. I'd need to know more about how their roving tab index works, if it kicks in after blur vs on tab key, etc.

However, according to their docs, they only mention tabbing through cells in the custom section, I think you could just turn off the Tab behavior https://www.ag-grid.com/react-data-grid/keyboard-navigation/#custom-navigation everything is navigable via the arrow keys already, and I checked that works in our dialog.

Edit: Looked again at our FocusScope with contain, it looks like we don't just prevent focus from leaving, we actually take over the focus movement for Tab completely https://github.com/adobe/react-spectrum/blob/a0f0604e6dc461cc50dd923b494b7ad9b9e5bd0c/packages/%40react-aria/focus/src/FocusScope.tsx#L341 which is why it's focusing directly onto the buttons in the table for example.

**Reply by p1pah** [reply]
@snowystinger any chance we could override this 😄 ?

**Reply by snowystinger** [reply, marked as answer]
you could probably stopPropagation of the Tab keyDown before it gets to the FocusScope, but I'd test when focus "leaves" the table

**Reply by p1pah** [reply]
@snowystinger you have restored my sleep schedule. 
![image](https://github.com/adobe/react-spectrum/assets/17172000/d4184cad-0a91-40f7-a8ab-8d2e669af6d4)
We will just wrap our grid divs with something like this. Thank you so much for the responsiveness and the help!
